spyder/app/tests/test_mainwindow.py

```diff
@@ -946,8 +955,10 @@ def test_shell_execution(main_window, qtbot, tmpdir):
 
 
 @flaky(max_runs=3)
-@pytest.mark.skipif(sys.platform.startswith('linux'),
-                    reason="Fails frequently on Linux")
+@pytest.mark.skipif(
+    sys.platform.startswith('linux') and running_in_ci(),
+    reason="Fails frequently on Linux"
```

Comment:
So this is working on linux locally but not on CI, right?

```suggestion
    sys.platform.startswith('linux') and running_in_ci(),
    reason="Fails frequently on Linux CI"
```